citron-collection-partitioner/collection_partitioner.py

```diff
@@ -0,0 +1,160 @@
+
+        keys = []
+
+        sts_client= CollectionPartitioner.init_sts_client()
+        assumed_role_object = sts_client.assume_role(
+            RoleArn="arn:aws:iam::585762237892:role/gads-citron-modeo",
+            RoleSessionName="AssumeRoleSession1",
+        )
+        credentials = assumed_role_object["Credentials"]
+        s3_client = boto3.client(
+            "s3",
+            aws_access_key_id=credentials["AccessKeyId"],
+            aws_secret_access_key=credentials["SecretAccessKey"],
+            aws_session_token=credentials["SessionToken"],
+        )
+
+        for file in s3_client.list_objects(Bucket=self.bucket, Prefix= collection_key)["Contents"]:
+            if not file["Key"].endswith('/'):
+                keys.append(file["Key"])
+
+        sts_client.close()
+        s3_client.close()
+        return keys
+
+    @staticmethod
+    def get_file_infos(key: str) -> dict:
+        """
+
+        :param key: key of the file stored in s3
+        :return: dict containing the information of the file: key, year, month, day
+        """
+
+        res= re.search(r"^.*year=(?P<year>\d+)/month=(?P<month>\d+)/day=(?P<day>\d+)/(?P<file_name>.*)$",key)
+        file_infos={
+            "key": key,
+            "file_name":res.group('file_name'),
+            "year":int(res.group('year')),
+            "month":int(res.group('month')),
+            "day":int(res.group('day')),
+        }
+        return file_infos
+
+    def create_collection_structure(self, collection_key: str) -> pd.DataFrame:
+        """
+
+        :param collection_key: key of the collection in s3
+        :return: return a pandas dataframe that summarizes the information of each file in this collection
+        """
+        data= pd.DataFrame(columns=["key","file_name","year","month","day"])
+        files= self.get_files_of_collection(collection_key)
+        for file in files:
+            data.loc[len(data.index)]= list(CollectionPartitioner.get_file_infos(file).values())
+
+        return data
+
+    def partition_collection(self, collection_key:str, year:bool, month:bool,day:bool,new_collection_key:str):
+        """
+
+        :param collection_key: existing collection key
+        :param year: bool to partition with year
+        :param month: bool to partition with month
+        :param day: bool to partition with day
+        :param new_collection_key: new partitioned collection key
+        :return: create a new partitioned collection from an existing collection depending on the passed params
+        """
+
+        data= self.create_collection_structure(collection_key)
+        partition_parameters=[]
+        if year: partition_parameters.append('year')
+        if month: partition_parameters.append('month')
+        if day: partition_parameters.append('day')
+
+        aws_client = self.init_sts_client()
+
+        assumed_role_object = aws_client.assume_role(
+            RoleArn="arn:aws:iam::585762237892:role/gads-citron-modeo",
```

Comment:
This STRING should be defined as a global variable at the top of the file : 

`CITRON_ROLE_ARN = arn:aws:iam::585762237892:role/gads-citron-modeo`